sdk/src/common/env_variables.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/common/env_variables.h"
+
+#ifdef _MSC_VER
+#  include <string.h>
+#  define strcasecmp _stricmp
+#else
+#  include <strings.h>
+#endif
+
+#include <memory>
+
+#include "opentelemetry/sdk/common/global_log_handler.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+
+bool GetRawEnvironmentVariable(const char *env_var_name, std::string &value)
+{
+#if !defined(NO_GETENV)
+  const char *endpoint_from_env = nullptr;
+#  if defined(_MSC_VER)
+  // avoid calling std::getenv which is deprecated in MSVC.
+  size_t required_size = 0;
+  getenv_s(&required_size, nullptr, 0, env_var_name);
+  std::unique_ptr<char> endpoint_buffer;
```

Comment:
Should it be `std::unique_ptr<char[]>`?

Reply:
Indeed, thanks.

This code comes from the main branch, so it should be fixed there also, to avoid keeping the fix in this PR for too long.

Please see PR #1971